core/navigation.js

```diff
@@ -497,6 +497,43 @@ Blockly.Navigation.insertAtConnection = function(newBlock, connection) {
   Blockly.Navigation.insertBlock(newBlock, connection);
 };
 
+/**
+ * Disconnect the connection that the cursor is pointing to, and bump blocks.
+ * This is a no-op if the connection cannot be broken or if the cursor is not
+ * pointing to a connection.
+ * @package
+ */
+Blockly.Navigation.disconnectBlocks = function() {
+  var curNode = Blockly.Navigation.cursor_.getCurNode();
+  if (!curNode.isConnection()) {
```

Comment:
Should we allow disconnecting if you're on a block and it has a prev/output connection?

Reply:
I don't think so--it's not obvious at that point which connection you're going for.